packages/by-name/qemu-tdx-static/0004-hw-x86-load-initrd-to-static-address.patch

```diff
@@ -0,0 +1,42 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moritz Sanft <58110325+msanft@users.noreply.github.com>
+Date: Thu, 21 Nov 2024 14:36:23 +0100
+Subject: [PATCH] hw/x86: load initrd to static address
+
+For TDX RTMRs to be predictable regardless of VM memory size, we need to
+load the initrd to a static address, so no dynamic value ends up in the
+mapped kernel image. (as the initrd address does)
+As we control the minimum VM memory size in Contrast, we just load the initrd
+to the address it gets loaded to for Contrast's minimum VM memory, regardless
+of if the VM has more memory, as it would otherwise happen with QEMU's 4Gi
+constraint.
```

Comment:
That 4Gi constraint is mentioned a few times, but I believe it's not explained anywhere.

Reply:
It's in the line above the comment, which is why I didn't point it out more here. Will add a direct reference instead.

Reply:
I tried to explain it in more detail now. Please let me know if it's more follow-able now.